pandas/tseries/tdi.py

```diff
@@ -440,21 +439,18 @@ def astype(self, dtype):
 
         if dtype == np.object_:
             return self.asobject
-        elif dtype == _INT64_DTYPE:
-            return self.asi8.copy()
         elif dtype == _TD_DTYPE:
```

Comment:
Shouldn't it be rather `is_timedelta64_ns_dtype(dtype)`?
BTW, this function is broken in `pandas/core/common.py` but there's an easy fix.